node/pkg/processor/message.go

```diff
@@ -93,7 +93,7 @@ func (p *Processor) handleMessage(ctx context.Context, k *common.MessagePublicat
 	}
 
 	// Broadcast the signature.
-	ourObs, msg := p.broadcastSignature(v.MessageID(), k.TxID, digest, signature, shouldPublishImmediately)
+	ourObs, msg := p.broadcastSignature(v.MessageID(), k.TxID, digest, signature, shouldPublishImmediately, k.EmitterChain)
```

Comment:
Thoughts on just passing `k` (MessagePublication) instead of `k.TxID` and `k.EmitterChain`?